packages/providers/src/transaction-summary/assemble-transaction-summary.ts

```diff
@@ -0,0 +1,133 @@
+import { hexlify } from '@ethersproject/bytes';
+import type { BN } from '@fuel-ts/math';
+import type { Transaction } from '@fuel-ts/transactions';
+
+import type { TransactionResultReceipt } from '../transaction-response';
+import { calculateTransactionFee } from '../utils';
+
+import {
+  getOperations,
+  getTransactionTypeName,
+  isTypeMint,
+  isTypeCreate,
+  isTypeScript,
+} from './operations';
+import type {
+  BlockId,
+  GqlTransactionStatusesNames,
+  GraphqlTransactionStatus,
+  Time,
+  TransactionSummary,
+} from './types';
+import { SimplifiedTransactionStatusNameEnum } from './types';
+
+export const getTransactionStatusName = (gqlStatus: GqlTransactionStatusesNames) => {
+  switch (gqlStatus) {
+    case 'FailureStatus':
+      return SimplifiedTransactionStatusNameEnum.failure;
+    case 'SuccessStatus':
+      return SimplifiedTransactionStatusNameEnum.success;
+    case 'SubmittedStatus':
+      return SimplifiedTransactionStatusNameEnum.submitted;
+    case 'SqueezedOutStatus':
+      return SimplifiedTransactionStatusNameEnum.squeezedout;
+    default:
+      throw new Error('Unknown transaction status');
+  }
+};
+
+export function assembleTransactionSummary<TTransactionType = void>(params: {
+  id?: string;
+  gasPrice: BN;
+  gasPerByte?: BN;
+  gasPriceFactor?: BN;
+  transaction: Transaction;
+  transactionBytes: Uint8Array;
+  gqlTransactionStatus?: GraphqlTransactionStatus;
+  receipts: TransactionResultReceipt[];
+}) {
+  const {
+    receipts,
+    gasPerByte,
+    gasPrice,
+    gasPriceFactor,
+    transaction,
+    transactionBytes,
+    id,
+    gqlTransactionStatus,
+  } = params;
+
+  const { gasUsed, fee } = calculateTransactionFee({
+    receipts,
+    gasPrice,
+    gasPerByte,
+    gasPriceFactor,
+    transactionBytes,
+    transactionType: transaction.type,
+    transactionWitnesses: transaction?.witnesses || [],
+  });
+
+  const operations = getOperations({
+    transactionType: transaction.type,
+    inputs: transaction.inputs || [],
+    outputs: transaction.outputs || [],
+    receipts,
+    rawPayload: hexlify(transactionBytes),
+    // abiMap, TODO: update this when it's possible to get a contract JsonAbi using graphql
```

Comment:
we should allow abiMap to be a input from user. if user has the contract ABI in hands we should accept